components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/SAPEndpoint.java

```diff
@@ -182,18 +184,23 @@ protected void setupOptionalFeatures(JCoServer server) throws Exception {
 
     /**
      * Block until server state is stopped or maximum timeout reached.
-     * @param server
-     * @return true | false if server state is JCoServerState.STOPPED in case timeout is exceeded
+     *
+     * @param server jco server for which we wait to be stopped.
+     * @return true if the server is stopped before the timeout is exceeded
      */
     protected boolean waitForServerStop(JCoServer server) {
         long timeStamp = System.currentTimeMillis();
-        do {
-            // This is to ensure a startEndpoint called right after a stopEndpoint (proxy redeploy) would not
-            // cause the server.start() to be called until the server state is STOPPED.
+        while (server.getState() != JCoServerState.STOPPED
+               && timeStamp + serverStopTimeout > System.currentTimeMillis()) {
             if (log.isDebugEnabled()) {
```

Comment:
Shall we move this outside the while loop?

Reply:
I think it's better to leave it inside the while loop since this gets executed only when trying to stop the jco server and when debug is enabled. Having the debug log inside would give us more insight as to the average time it takes for the JCo server to be stopped. WDYT?

Reply:
But that will print multiple of these lines, isn't it?

Reply:
Yes, but this is a debug log. Would be beneficial for tracking the time it took for stopping the server.